moirae/estimators/online/filters/distributions.py

```diff
@@ -43,12 +45,48 @@ def combine_with(self, random_dists: Iterable[Self]) -> Self:
         """
         raise NotImplementedError('Please implement in child class!')
 
+    @abstractmethod
+    def convert(self, conversion_operator: ConversionOperator) -> Self:
+        """
+        Uses the methods available in the :class:`~moirae.estimators.online.filters.transformations.BaseTransform` to
+        transform the underlying distribution and return a copy of the transformed
+        :class:`~moirae.estimators.online.filters.distributions.MultivariateRandomDistribution`
+
+        Args:
+            transform_operator: operator to perform necessary transformations
 
-class DeltaDistribution(MultivariateRandomDistribution):
-    """A distribution with only one allowed value"""
+        Returns:
+            transformed_dist: transformed distribution
+        """
+        raise NotImplementedError('Please implement in child class!')
+
+
+class DeltaDistribution(MultivariateRandomDistribution, validate_assignment=True):
+    """
+    A distribution with only one set of allowed values
+
+    Args:
+        mean: a 1D array containing allowed values; can be passed as flattened array or array with shapes (1, dim) or
+            (dim, 1)
+    """
 
     mean: np.ndarray = Field(default=None, description='Mean of the distribution.')
 
+    @field_validator('mean', mode='after')
+    @classmethod
+    def mean_1d(cls, mu: np.ndarray) -> np.ndarray:
+        """ Making sure the mean is a vector """
+        mean_shape = mu.shape
+        if not mean_shape:
+            raise ValueError('Mean must be Sized and have a non-empty shape!')
+        if len(mean_shape) > 2 or (len(mean_shape) == 2 and 1 not in mean_shape):
+            raise ValueError('Mean must be a 1D vector, but array provided has shape ' + str(mean_shape) + '!')
+        elif len(mean_shape) == 2:
```

Comment:
Why not throw an error when a user provides an incompatible value? Adding more "implicit" operations like this type conversion below makes the code more complicated without specific benefits.

Reply:
I added here because, in the model-related objects (`HealthVariable` and `GeneralContainer`), things are specified as 2D arrays, even when they are really 1D, and I didn't want to end up having a situation where everything breaks because somewhere we forgot to add a flatten. Based on this, do you still think I should throw an error here?

Reply:
Yes, so long as we make it clear what they did wrong. Maybe even add "did you forget "flatten"?"

Reply:
Gotcha, taking care of it!